src/dali_executor/dali_executor.cc

```diff
@@ -136,6 +134,32 @@ bool DaliExecutor::IsNoCopy(device_type_t es_device, const IDescr& input) {
           input.buffers[0].device_id == pipeline_.DeviceId());
 }
 
+
+static bool streq(const char * lhs, const char * rhs) {
+  return strcmp(lhs, rhs) == 0;
+}
+
+
+bool DaliExecutor::IsInputConsumed() {
+  for (auto &name : input_names_) {
+    auto trace = pipeline_.TryGetOperatorTrace(name, "depleted");
+    if (!trace.has_value()) {
+      throw std::runtime_error("DALI Error: 'depleted' trace shall be available for every input operator.");
+    }
+    if (streq(trace->c_str(), "true")) {
+      return true;
+    }
+  }
+  for (auto &name : input_names_) {
+    auto trace = pipeline_.TryGetOperatorTrace(name, "next_output_data_id");
+    if (trace.has_value() && std::stoull(*trace) != request_id_) {
```

Comment:
It'd be better to keep the current ID as a string and compare strings.
The code, as is, would say that any non-number matches `request_id_ == 0`.